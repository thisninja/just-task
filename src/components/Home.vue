<script>
export default {
  name: 'Home',
  render(h) {
    return h('div', { class: 'full-width' });
  },
}
</script>
