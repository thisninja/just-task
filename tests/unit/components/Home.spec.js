import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Home from '@/components/Home.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Home.vue', () => {
  let wrapper;
  let store;
  let actions;
  let mutations;
  let getters;

  beforeEach(() => {
    getters = getters || {
      isAuthenticated: () => null,
    };

    store = new Vuex.Store({
      getters,
    });

    wrapper = shallowMount(Home, {
      store,
      localVue,
    });
  });

  afterEach(() => {
    getters = null;
  });

  describe('snapshots', () => {
    describe('!isAuthenticated', () => {
      it('should match snapshot if !isAuthenticated', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe('isAuthenticated', () => {
      beforeAll(() => {
        getters = {
          isAuthenticated: () => true
        }
      });

      it('should match snapshot if isAuthenticated true', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
