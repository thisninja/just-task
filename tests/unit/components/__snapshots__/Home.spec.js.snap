// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue snapshots !isAuthenticated should match snapshot if !isAuthenticated 1`] = `
<div class="full-width">
  <login-stub></login-stub>
</div>
`;

exports[`Home.vue snapshots isAuthenticated should match snapshot if isAuthenticated true 1`] = `
<div class="full-width">
  <task-stub></task-stub>
</div>
`;
